Portal/src/Datahub.Infrastructure/Services/Projects/ProjectUsageService.cs

```diff
@@ -180,8 +180,13 @@ private List<AzureServiceCost> GetTotalCostsPerService(List<AzureServiceCost> co
             var existing = totalCosts.FirstOrDefault(c => c.Name == cost.Name);
             if (existing is null)
             {
-                cost.Date = DateTime.MinValue;
-                totalCosts.Add(cost);
+                var newCost = new AzureServiceCost
```

Comment:
I think you could replace the whole function with

return costs.GroupBy(c => c.Name).Select(gp => new AzureServiceCost{Date = gp.Min(g => g.Date), Name = gp.Key.Name, Cost = gp.Sum(g => g.Cost)).ToList()

Can you have a look - I think we should make sure the logic here is clear and easy to read.